File: Joints.m
function Joints
    clc;
    values = InputMoment();
    FX = values(1);
    FY = values(2);
    MomentDueToForce = values(3);
    LoadValues = InputLoad();
    LX = LoadValues(1);
    LY = LoadValues(2);
    MomentDueToLoad = LoadValues(3);
    TotalFX = FX + LX;
    TotalFY = FY + LY;
    
    choice = input('If any extra moments are present enter 1 else 0: ');
    ExtraMoment = 0;
    if choice == 1
        ExtraMoment = InputExtraMoments();
    end
    
    TotalMoment = MomentDueToForce + MomentDueToLoad + ExtraMoment;
    disp('Choose the System of Joints');
    disp('1. Roller and Hinge');
    disp('2. 2 Roller');
    disp('3. 2 Hinge [also works for 1 hinge and an inclined roller]');
    system = input('');
    switch system
        case 1
            Roller_Hinge(TotalFY, TotalFX, TotalMoment);
        case 2
            TwoRoller(TotalFY, TotalFX, TotalMoment);
        case 3
            TwoHinge(TotalFY, TotalFX, TotalMoment);
        otherwise
            disp('Invalid System of Joints');
    end
end

function values = InputMoment()
    clc;
    n = input('Enter the number of known forces: ');
    FX = 0;
    FY = 0;
    Moment = 0;
    angle = 0;
    Inclination = 2;
    for i = 1:n
        clc;
        fprintf('Enter the information of force %d\n', i);
        Inclination = input('Enter 1 if the force is inclined else 0: ');
        if Inclination == 1
            quadrant = InputQuadrant();
            magnitude = InputMagnitude();
            inputType = input('Select A for angle or S for slope: ', 's');
            switch inputType
                case {'A', 'a'}
                    angle = InputAngle(quadrant);
                case {'S', 's'}
                    angle = InputSlope(quadrant);
                otherwise
                    disp('Invalid Input');
            end
        else
            magnitude = InputMagnitude();
            disp('Choose the orientation of the force');
            disp('1. Upwards');
            disp('2. Downwards');
            disp('3. Rightwards');
            disp('4. Leftwards');
            direction = input('');
            if direction == 1
                angle = pi/2;
            elseif direction == 2
                angle = 3*pi/2;
            elseif direction == 3
                angle = 0;
            elseif direction == 4
                angle = pi;
            end
        end
        FX = FX + CalcFX(magnitude, angle);
        FY = FY + CalcFY(magnitude, angle);
        
        distance = input('Enter the distance from the reference point: ');
        disp('Choose the direction of moment caused by the force');
        disp('1. Anti-clockwise');
        disp('2. Clockwise');
        disp('0. If the Force is passing through the point');
        direction = input('');
        Moment = Moment + CalcMoment(magnitude, distance, direction);
    end
    values = [FX, FY, Moment];
end

function Moment = InputExtraMoments()
    clc;
    n = input('Enter the number of moments: ');
    Moment = 0;
    for i = 1:n
        magnitude = input(['Enter the Magnitude of the Moment ', num2str(i), ': ']);
        direction = input('Select 1 for Moment acting Anticlockwise or 2 for Clockwise: ');
        if direction == 1
            Moment = Moment + magnitude;
        elseif direction == 2
            Moment = Moment - magnitude;
        else
            disp('Invalid direction');
        end
    end
end

function values = InputLoad()
    n = input('Enter the number of loads: ');
    Load = zeros(1, 100);
    Distance = zeros(1, 100);
    Direction = zeros(1, 100);
    for i = 1:n
        clc;
        disp('Select the type of load');
        disp('1 for Point Load');
        disp('2 for Uniform Distributed Load');
        disp('3 for Uniformly Varying Load');
        disp('4 for Combined UDL and UVL');
        choice = input('');
        switch choice
            case 1
                Load(i) = input('Enter the magnitude of the Point Load: ');
                Distance(i) = input('Enter the distance from the reference point: ');
                disp('Choose the direction of Potential Moment');
                disp('1: Anticlockwise');
                disp('2: Clockwise');
                Direction(i) = input('');
            case 2
                magnitude = input('Enter the magnitude of the Load: ');
                length = input('Enter the length of applied Load: ');
                distance = input('Enter the distance of the closer side from the reference point: ');
                disp('Choose the direction of Potential Moment');
                disp('1: Anticlockwise');
                disp('2: Clockwise');
                Direction(i) = input('');
                Load(i) = magnitude * length;
                Distance(i) = distance + (length / 2);
            case 3
                disp('Choose the orientation of the UVL');
                disp('1: Slanted toward the reference point');
                disp('2: Slanted away from the reference point');
                orientation = input('');
                magnitude = input('Enter the magnitude of the Load: ');
                length = input('Enter the length of applied Load: ');
                distance = input('Enter the distance of the closer side from the reference point: ');
                disp('Choose the direction of Potential Moment');
                disp('1: Anticlockwise');
                disp('2: Clockwise');
                Direction(i) = input('');
                Load(i) = 0.5 * magnitude * length;
                if orientation == 1
                    Distance(i) = distance + (2 * length) / 3;
                elseif orientation == 2
                    Distance(i) = distance + length / 3;
                end
            case 4
                disp('Choose the orientation of the UVL');
                disp('1: Slanted toward the reference point');
                disp('2: Slanted away from the reference point');
                orientation = input('');
                smallermagnitude = input('Enter the smaller magnitude of the Load: ');
                biggermagnitude = input('Enter the bigger magnitude of the Load: ');
                length = input('Enter the length of applied Load: ');
                distance = input('Enter the distance of the closer side from the reference point: ');
                disp('Choose the direction of Potential Moment');
                disp('1: Anticlockwise');
                disp('2: Clockwise');
                Direction(i) = input('');
                Load(i) = smallermagnitude * length;
                Distance(i) = distance + (length / 2);
                
                Load(i + 1) = 0.5 * (biggermagnitude - smallermagnitude) * length;
                if orientation == 1
                    Distance(i + 1) = distance + (2 * length) / 3;
                elseif orientation == 2
                    Distance(i + 1) = distance + length / 3;
                end
                Direction(i + 1) = Direction(i);
                i = i + 1;
                n = n + 1;
            otherwise
                disp('Invalid Input');
        end
    end
    
    LX = 0;
    LY = 0;
    for i = 1:n
        LY = LY + CalcFY(Load(i), deg2rad(270));
    end
    
    MomentDueToLoad = 0;
    for i = 1:n
        MomentDueToLoad = MomentDueToLoad + CalcMoment(Load(i), Distance(i), Direction(i));
    end
    
    values = [LX, LY, MomentDueToLoad];
end

function Roller_Hinge(FY, FX, Moment)
    clc;
    disp('Using the Hinge Joint as reference point');
    RollerValues = InputJoint('Roller');
    RollerDistance = RollerValues(1);
    RollerOrientation = RollerValues(2);
    HingeValues = InputJoint('Hinge');
    HingeDistance = HingeValues(1);
    HingeOrientation = HingeValues(2);
    
    ReactionValues = simultaneousEqnSolver(RollerDistance, HingeDistance, -Moment, 1, 1, -FY);
    RollerReaction = ReactionValues(1);
    HingeReactionFY = ReactionValues(2);
    HingeReactionFX = -FX;
    HingeReaction = hypot(HingeReactionFX, HingeReactionFY);
    
    disp(['Roller Reaction: ', num2str(RollerReaction), ' N']);
    disp(['Hinge Reaction: ', num2str(HingeReaction), ' N']);
    disp('"Approx Values"');
end

function TwoRoller(FY, FX, Moment)
    clc;
    Roller1Values = InputJoint('Roller');
    Roller1Distance = Roller1Values(1);
    Roller1Orientation = Roller1Values(2);
    Roller2Values = InputJoint('Roller');
    Roller2Distance = Roller2Values(1);
    Roller2Orientation = Roller2Values(2);
    
    ReactionValues = simultaneousEqnSolver(Roller1Distance, Roller2Distance, -Moment, 1, 1, -FY);
    Roller1Reaction = ReactionValues(1);
    Roller2Reaction = ReactionValues(2);
    
    disp(['Roller 1 Reaction: ', num2str(Roller1Reaction), ' N']);
    disp(['Roller 2 Reaction: ', num2str(Roller2Reaction), ' N']);
    disp('"Approx Values"');
end

function TwoHinge(FY, FX, Moment)
    clc;
    Hinge1Values = InputJoint('Hinge');
    Hinge1Distance = Hinge1Values(1);
    Hinge1Orientation = Hinge1Values(2);
    Hinge2Values = InputJoint('Hinge');
    Hinge2Distance = Hinge2Values(1);
    Hinge2Orientation = Hinge2Values(2);
    
    ReactionValues = simultaneousEqnSolver(Hinge1Distance, Hinge2Distance, -Moment, 1, 1, -FY);
    Hinge1Reaction = ReactionValues(1);
    Hinge2Reaction = ReactionValues(2);
    
    disp(['Hinge 1 Reaction: ', num2str(Hinge1Reaction), ' N']);
    disp(['Hinge 2 Reaction: ', num2str(Hinge2Reaction), ' N']);
    disp('"Approx Values"');
end

function distance = InputDistance()
    distance = input('Enter the distance from the reference point: ');
end

function magnitude = InputMagnitude()
    magnitude = input('Enter the magnitude of the force: ');
end

function angle = InputAngle(quadrant)
    angle = input('Enter the value of the angle in degrees: ');
    switch quadrant
        case 1
            angle = angle;
        case 2
            angle = 180 - angle;
        case 3
            angle = 180 + angle;
        case 4
            angle = 360 - angle;
        otherwise
            disp('Invalid Quadrant');
    end
end

function angle = InputSlope(quadrant)
    SlopeRatio = zeros(2, 1);
    SlopeRatio(1) = input('Enter the value of X Component: ');
    SlopeRatio(2) = input('Enter the value of Y Component: ');
    switch quadrant
        case 1
            angle = atan(SlopeRatio(2) / SlopeRatio(1));
        case 2
            angle = pi - atan(SlopeRatio(2) / SlopeRatio(1));
        case 3
            angle = pi + atan(SlopeRatio(2) / SlopeRatio(1));
        case 4
            angle = 2*pi - atan(SlopeRatio(2) / SlopeRatio(1));
        otherwise
            disp('Invalid Quadrant');
    end
end

function quadrant = InputQuadrant()
    disp('Choose the Quadrant of the angle:');
    disp('1: First Quadrant');
    disp('2: Second Quadrant');
    disp('3: Third Quadrant');
    disp('4: Fourth Quadrant');
    quadrant = input('');
end

function values = InputJoint(jointType)
    clc;
    fprintf('Enter the Information of %s joint\n', jointType);
    distance = InputDistance();
    disp('Choose the orientation of the joint');
    disp('1. Horizontal');
    disp('2. Vertical');
    orientation = input('');
    values = [distance, orientation];
end

function fx = CalcFX(magnitude, angle)
    fx = magnitude * cos(angle);
end

function fy = CalcFY(magnitude, angle)
    fy = magnitude * sin(angle);
end

function moment = CalcMoment(magnitude, distance, direction)
    if direction == 1
        moment = magnitude * distance;
    elseif direction == 2
        moment = -magnitude * distance;
    else
        moment = 0;
    end
end

function solutions = simultaneousEqnSolver(coeff1, coeff2, rhs1, coeff3, coeff4, rhs2)
    A = [coeff1, coeff2; coeff3, coeff4];
    B = [rhs1; rhs2];
    solutions = linsolve(A, B);
end

File: Joints.java
import java.util.*;
import java.lang.*;
import java.text.*;

public class Joints
{
    public static void main(String[] args)
    {
        Joints();
    }
    public static void Joints()
    {
        Scanner sc = new Scanner(System.in) ;
        double[] values = InputMoment();
        double FX = values[0];
        double FY = values[1];
        double MomentDueToForce = values[2];
        double[] LoadValues = InputLoad();
        double LX = LoadValues[0];
        double LY = LoadValues[1];
        double MomentDueToLoad = LoadValues[2];
        double TotalFX = FX + LX;
        double TotalFY = FY + LY;
        System.out.println("If any extra moments are present enter 1 else 0");
        int choice = sc.nextInt();
        double ExtraMoment = 0 ;
        if(choice == 1)
        {
            ExtraMoment = InputExtraMoments();
        }
        double TotalMoment = MomentDueToForce + MomentDueToLoad + ExtraMoment ;
        System.out.println("Choose the System of Joints");
        System.out.println("1. Roller and Hinge");
        System.out.println("2. 2 Roller");
        System.out.println("3. 2 Hinge [also works for 1 hinge and an inclined roller]");
        int system = sc.nextInt();
        switch(system)
        {
            case 1:
            Roller_Hinge(TotalFY ,  TotalFX , TotalMoment);
            break;
            case 2:
            TwoRoller(TotalFY , TotalFX , TotalMoment);
            break;
            case 3:
            TwoHinge(TotalFY , TotalFX , TotalMoment);
            break;
            default:
            System.out.println("Invalid System of Joints");
            break;
        }

    }
    public static double[] InputMoment()
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("\033[H\033[2J");
        System.out.println("Enter the number of known forces :");
        int n = sc.nextInt() ;
        double FX=0 , FY=0 ;
        double Moment = 0 ;
        double angle =0 ;
        int Inclination = 2;
        for( int i = 0 ; i < n ; i++)
        {
            System.out.println("\033[H\033[2J");
            System.out.println("Enter the information of force" + (i+1));
            System.out.println("Enter 1 if the force is inclined else 0");
            Inclination = sc.nextInt();
            double magnitude;
            if(Inclination == 1)
            {
                int quadrant = InputQuadrant();
                magnitude = InputMagnitude();            
                System.out.println("Select A for angle or S for slope");
                char input = sc.next().charAt(0);
                switch (input)
                {
                    case 'A':
                        angle = InputAngle(quadrant);
                        break;
                    case 'S':
                        angle = InputSlope(quadrant);
                        break;
                    case 'a':
                        angle = InputAngle(quadrant);
                        break;
                    case 's':
                        angle = InputSlope(quadrant);
                        break;
                    default:
                        System.out.println("Invalid Input");
                        break;
                }
            }
            else
            {
                magnitude = InputMagnitude();
                System.out.println("Choose the orientation of the force");
                System.out.println("1. Upwards");
                System.out.println("2. Downwards");
                System.out.println("3. Rightwards");
                System.out.println("4. Leftwards");
                int direction = sc.nextInt();
                if(direction == 1)
                {
                    angle = Math.PI/2 ;
                }
                if(direction == 2)
                {
                    angle = (3*Math.PI)/2 ;
                }
                if(direction == 3)
                {
                    angle = 0;  
                }
                if(direction == 4)
                {
                    angle = Math.PI ;
                }
            FX += CalcFX(magnitude , angle);
            FY += CalcFY(magnitude , angle);

            System.out.println("Enter the distance from the reference point");
            double distance = sc.nextDouble() ;
            System.out.println("Choose the direction of moment caused by the force");
            System.out.println("1. Anti-clockwise");
            System.out.println("2. Clockwise");
            System.out.println("0. If the Force is passing through the point");
            direction = sc.nextInt() ;
            Moment += CalcMoment(magnitude , distance , direction);
            }
        }
        double[] Finalvalues = new double[3];
        Finalvalues[0] = FX ;
        Finalvalues[1] = FY ;
        Finalvalues[2] = Moment ;
        return Finalvalues ;
    }
    public static double InputExtraMoments()
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("\033[H\033[2J");
        System.out.println("Enter the number of moments :");
        int n = sc.nextInt() ;
        double Moment = 0 ;
        int direction;
        for( int i = 0 ; i < n ; i++)
        {
            System.out.println("Enter the Magnitude of the Moment " + (i+1));
            double magnitude = sc.nextDouble() ;
            System.out.println("Select 1 for Moment acting Anticlockwise or 2 for Clockwise");
            direction = sc.nextInt() ;
            switch(direction)
            {
                case 1:
                Moment += magnitude ;
                break;
                case 2:
                Moment -= magnitude ;
                break;
                default :
                System.out.println("Invalid direction");
                break;
            }
        }
        return Moment ;
    }
    public static double[] InputLoad()
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("Enter the number of loads");
        int n = sc.nextInt() ;
        double[] Load = new double[100];
        double[] Distance = new double[100];
        int[] Direction = new int[100];
        for(int i = 0 ; i < n ; i++)
        {
            System.out.println("\033[H\033[2J");
            System.out.println("Select the type of load");
            System.out.println("1 for Point Load");
            System.out.println("2 for Uniform Distributed Load");
            System.out.println("3 for Uniformly Varying Load");
            System.out.println("4 for Combined UDL and UVL");
            int choice = sc.nextInt();
            double magnitude;
            double length;
            double smallermagnitude;
            double biggermagnitude;
            double distance;
            int orientation ;
            switch(choice)
            {
                case 1:
                System.out.println("Enter the magnitude of the Point Load");
                Load[i] = sc.nextDouble();
                System.out.println("Enter the distance from the reference point");
                Distance[i]= sc.nextDouble();
                System.out.println("Choose the direction of Potential Moment");
                System.out.println("1: Anticlockwise");
                System.out.println("2: Clockwise");
                Direction[i] = sc.nextInt();
                break;

                case 2:
                System.out.println("Enter the magnitude of the Load");
                magnitude = sc.nextDouble();
                System.out.println("Enter the length of applied Load");
                length = sc.nextDouble();
                System.out.println("Enter the distance of the closer side from the reference point");
                distance = sc.nextDouble();
                System.out.println("Choose the direction of Potential Moment");
                System.out.println("1: Anticlockwise");
                System.out.println("2: Clockwise");
                Direction[i] = sc.nextInt();
                Load[i] = magnitude*length;
                Distance[i] = distance + (length/2);
                break;

                case 3:
                System.out.println("Choose the orientation of the UVL");
                System.out.println("1: Slanted toward the reference point");
                System.out.println("2: Slanted away from thr reference point");
                orientation = sc.nextInt();
                System.out.println("Enter the magnitude of the Load");
                magnitude = sc.nextDouble();
                System.out.println("Enter the length of applied Load");
                length = sc.nextDouble();
                System.out.println("Enter the distance of the closer side from the reference point");
                distance = sc.nextDouble();
                System.out.println("Choose the direction of Potential Moment");
                System.out.println("1: Anticlockwise");
                System.out.println("2: Clockwise");
                Direction[i] = sc.nextInt();
                Load[i] = (0.5*magnitude*length);
                if(orientation ==1)
                {
                    Distance[i] = distance + ((2*length)/3);
                }
                if(orientation == 2)
                {
                    Distance[i] = distance + (length/3);
                }
                break;

                case 4:
                System.out.println("Choose the orientation of the UVL");
                System.out.println("1: Slanted toward the reference point");
                System.out.println("2: Slanted away from thr reference point");
                orientation = sc.nextInt();
                System.out.println("Enter the smaller magnitude of the Load");
                smallermagnitude = sc.nextDouble();
                System.out.println("Enter the bigger magnitude of the Load");
                biggermagnitude = sc.nextDouble();
                System.out.println("Enter the length of applied Load");
                length = sc.nextDouble();
                System.out.println("Enter the distance of the closer side from the reference point");
                distance = sc.nextDouble();
                System.out.println("Choose the direction of Potential Moment");
                System.out.println("1: Anticlockwise");
                System.out.println("2: Clockwise");
                Direction[i] = sc.nextInt();
                Load[i] = smallermagnitude*length;
                Distance[i] = distance + (length/2);

                Load[i+1] = (0.5*(biggermagnitude-smallermagnitude)*length);
                if(orientation ==1)
                {
                    Distance[i+1] = distance + ((2*length)/3);
                }
                if(orientation == 2)
                {
                    Distance[i+1] = distance + (length/3);
                }
                Direction[i+1]= Direction[i];
                i++;
                n++;
                break;
                default:
                System.out.println("Invalid Input");
                break;
            }
        }
        double LX = 0;
        double LY = 0;
        for(int i = 0 ; i < n ; i++)
        {
            LY += CalcFY(Load[i], Math.toRadians(270));
        }
        double MomentDueToLoad = 0 ;
        for(int i = 0 ; i < n ; i++)
        {
            MomentDueToLoad += CalcMoment(Load[i] , Distance[i] , Direction[i]);
        }

        double[] FinalValues = new double[3];
        FinalValues[0] = LX;
        FinalValues[1] = LY;
        FinalValues[2] = MomentDueToLoad;
        return FinalValues;
    }
    public static void Roller_Hinge(double FY , double FX , double Moment)
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("\033[H\033[2J");
        System.out.println("Using the Hinge Joint as reference point");
        double[] RollerValues = InputJoint("Roller");
        double RollerDistance = RollerValues[0];
        double RollerOrientation = RollerValues[1];
        double[] HingeValues = InputJoint("Hinge");
        double HingeDistance = HingeValues[0];
        double HingeOrientation = HingeValues[1];
        
        double[] ReactionValues = simultaneousEqnSolver(RollerDistance,HingeDistance,(0-Moment),1,1,(0-FY));
        double RollerReaction = ReactionValues[0];
        double HingeReactionFY = ReactionValues[1];
        double HingeReactionFX = 0 - FX;
        double HingeReaction = Math.hypot(HingeReactionFX, HingeReactionFY);

        System.out.println("Roller Reaction: " + RollerReaction + " N");
        System.out.println("Hinge Reaction: " + HingeReaction + " N");
        System.out.println("\"Aproxx Values\"");

    }
    public static void TwoRoller(double FY , double FX , double Moment)
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("\033[H\033[2J");
        double[] Roller1Values = InputJoint("Roller");
        double Roller1Distance = Roller1Values[0];
        double Roller1Orientation = Roller1Values[1];
        double[] Roller2Values = InputJoint("Roller");
        double Roller2Distance = Roller2Values[0];
        double Roller2Orientation = Roller2Values[1];
        
        double[] ReactionValues = simultaneousEqnSolver(Roller1Distance,Roller2Distance,(0-Moment),1,1,(0-FY));
        double Roller1Reaction = ReactionValues[0];
        double Roller2Reaction = ReactionValues[1];

        System.out.println("Roller 1 Reaction : " +Roller1Reaction+ " N");
        System.out.println("Roller 2 Reaction : " + Roller2Reaction+ " N");
        System.out.println("\"Aproxx Values\"");
    }
    public static void TwoHinge(double FY , double FX , double Moment)
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("\033[H\033[2J");
        System.out.println("Hinge Joint 1");
        double[] Hinge1Values = InputJoint("Hinge");
        double Hinge1Distance = Hinge1Values[0];
        double Hinge1Orientation = Hinge1Values[1];
        System.out.println("Hinge Joint 2");
        double[] Hinge2Values = InputJoint("Hinge");
        double Hinge2Distance = Hinge2Values[0];
        double Hinge2Orientation = Hinge1Values[1];

        double[] ReactionValues = simultaneousEqnSolver(Hinge1Distance,Hinge2Distance,(0-Moment),1,1,(0-FY));
        double Hinge1Reaction = ReactionValues[0];
        double Hinge2Reaction = ReactionValues[1];

        System.out.println("Hinge 1 Reaction : " + Hinge1Reaction+ " N");
        System.out.println("Hinge 2 Reaction : " + Hinge2Reaction+ " N");
        System.out.println("\"Approx Values\"");
        System.out.println("IF USED FOR INCLINED ROLLER ENTER 1");
        int inclinedRoller = sc.nextInt();
        if(inclinedRoller == 1)
        {
            System.out.println("Select the hinge used as the inclined roller");
            System.out.println("1. Hinge 1");
            System.out.println("2. Hinge 2");
            int hinge = sc.nextInt();
            double roller= 0;
            if(hinge == 1)
            {
                roller = Hinge1Reaction;
            }
            if(hinge == 2)
            {
                roller = Hinge2Reaction;
            }
            System.out.println("Enter the angle of inclination with the positive y-axis");
            double angle = sc.nextDouble();
            roller = roller *Math.cos(angle);
            System.out.println("Roller Reaction : " + roller + " N");
        }
    }
    public static double[] InputJoint(String name)
    {
        Scanner sc = new Scanner(System.in) ;
        double distance=0;
        double orientation=0;
        switch(name)
        {
            case "Roller":  
            System.out.println("Enter the Distance of the Roller from the reference Point");
            System.out.println("Enter 0 if it is reference point");
            distance = sc.nextDouble() ;
            System.out.println("Select the orientation of moment caused by the joint");
            System.out.println("1. Anti-Clockwise");
            System.out.println("2.Clockwise");
            System.out.println("0. If joint is at reference point");
            orientation = sc.nextDouble();
            break;
            case "Hinge":
            System.out.println("Enter the Distance of the Hinge from the reference Point");
            System.out.println("Enter 0 if it is reference point");
            distance = sc.nextDouble() ;
            System.out.println("Select the orientation of moment caused by the joint");
            System.out.println("1. Anti-Clockwise");
            System.out.println("2.Clockwise");
            System.out.println("0. If joint is at reference point");
            orientation = sc.nextDouble();
            break;
            case "Fixed":
            System.out.println("Enter the Distance of the Fixed from the reference Point");
            System.out.println("Enter 0 if it is reference point");  
            distance = sc.nextDouble() ;
            System.out.println("Select the orientation of moment caused by the joint");
            System.out.println("1. Anti-Clockwise");
            System.out.println("2.Clockwise");
            System.out.println("0. If joint is at reference point");
            orientation = sc.nextDouble();
            break;
            default:
            System.out.println("Invalid Joint");
            break;
        }

        double[] JointValues = new double[2];
        JointValues[0] = distance;
        JointValues[1] = orientation;
        return JointValues;
    }
    public static int InputQuadrant()
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("Select the quadrant the force is present in : ");
        System.out.println("[Once the base of the force is at origin ]");
        System.out.println("1. Quadrant 1") ;
        System.out.println("2. Quadrant 2") ;
        System.out.println("3. Quadrant 3") ;
        System.out.println("4. Quadrant 4") ;
        int quadrant = sc.nextInt() ;
        return quadrant ;

    }
    public static double InputMagnitude()
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("Enter the magnitude of force :");
        double magnitude = sc.nextDouble() ;
        return magnitude ;
    }
    public static double InputAngle(int quadrant)
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("Enter angle between the force and the respective X-Axis :");
        double angle = sc.nextDouble() ;
        if(quadrant == 1)
        {
            angle = Math.toRadians(angle) ;
        }
        if(quadrant == 2)
        {
            angle = Math.PI-Math.toRadians(angle) ;
        }
        if(quadrant == 3)
        {
            angle = Math.PI+Math.toRadians(angle) ;
        }
        if(quadrant == 4)
        {
            angle = (2*Math.PI)-Math.toRadians(angle) ;
        }
        return angle ;
    }
    public static double InputSlope(int quadrant)
    {
        Scanner sc = new Scanner(System.in) ;
        System.out.println("Enter the horizontal length for the slope :");
        double horizontal = sc.nextDouble() ;
        System.out.println("Enter the vertical length for the slope :");
        double vertical = sc.nextDouble() ;

        double angle = Math.atan(vertical/horizontal) ;
        if(quadrant == 1)
        {
            angle = (angle) ;
        }
        if(quadrant == 2)
        {
            angle = Math.PI - (angle) ;
        }
        if(quadrant == 3)
        {
            angle = Math.PI + (angle) ;
        }
        if(quadrant == 4)
        {
            angle = (2*Math.PI) - (angle) ;
        }
        return angle ;
    }
    public static double CalcFX(double magnitude , double angle)
    {
        double FX = magnitude*Math.cos(angle);
        return FX ;
    }
    public static double CalcFY(double magnitude , double angle)
    {
        double FY = magnitude*Math.sin(angle);
        return FY ;
    }
    public static double[] simultaneousEqnSolver(double CoefX1 , double CoefY1 , double Const1 , double CoefX2 , double CoefY2 , double Const2)
    {
        //Coefx1(x) + Coefy1(y) = Const1
        //Coefx2(x) + Coefy2(y) = Const2
        double[] ResultantValues = new double[2];
        double x = (Const1*CoefY2 - Const2*CoefY1)/(CoefX1*CoefY2 - CoefX2*CoefY1);
        double y = (Const1*CoefX2 - Const2*CoefX1)/(CoefX2*CoefY1 - CoefX1*CoefY2);
        ResultantValues[0] = x ;
        ResultantValues[1] = y ;
        return ResultantValues ;
    }
    public static double CalcMoment(double magnitude , double distance , int direction)
    {
        if(direction == 1)
        {
            double Moment = magnitude*distance;
            return Moment ;
        }
        else if(direction == 2)
        {
            double Moment = -magnitude*distance;
            return Moment ;
        }
        else
        return 0 ;
    }
}